ngraph/frontend/onnx_import/src/core/graph.cpp

```diff
@@ -87,10 +87,22 @@ namespace ngraph
                 if (initializer_tensor.has_name())
                 {
                     Tensor tensor = Tensor{initializer_tensor};
-                    initializers.emplace(initializer_tensor.name(), tensor);
-
+                    std::shared_ptr<default_opset::Constant> ng_constant;
                     // For each initializer create a Constant node and store it in cache
-                    auto ng_constant = tensor.get_ng_constant();
+                    try
+                    {
+                        ng_constant = tensor.get_ng_constant();
+                    }
+                    catch (const ngraph::ngraph_error& exc)
+                    {
+                        NGRAPH_WARN << "Could not create an nGraph Constant for initializer '"
+                                    << initializer_tensor.name() << "'. Detailed error:\n"
+                                    << exc.what();
+                        ng_constant =
+                            default_opset::Constant::create(tensor.get_ng_type(), Shape{}, {0});
```

Comment:
Am I right that you create a scalar constant with 0? Can it be a root case of invalid networks after the conversion in the future?

Reply:
This path will only be executed if the initial ONNX model contains initializers(weights) with defined shape but no data. This is when an exception is thrown from the `get_ng_constant()` method that tries to extract the initializer's data.
Such situation was discovered when processing models that use empty initializers as optional inputs (for example in the Resize operator with 4 inputs, where the third should be ignored). To be able to handle such models and create a correct ngraph::Function, I had to pass something as an input to this op. The Resize op implementation doesn't use the data from this input though so the whole thing works as expected.
I'm not aware of any model that can cause problems with this approach right now but potentially we might encounter an error in the future.